gc.c

```diff
@@ -2761,6 +2767,23 @@ rb_gc_copy_attributes(VALUE dest, VALUE obj)
     rb_gc_impl_copy_attributes(rb_gc_get_objspace(), dest, obj);
 }
 
+const char *
+rb_gc_active_gc_name(void)
+{
+    const char *gc_name = rb_gc_impl_active_gc_name();
```

Comment:
Can we make `rb_gc_impl_active_gc_name` return a string literal or a malloc'd string? That way we don't need to allocate a new string every time.